### ISSUE TITLE
Issue with Crossover Calculations

### ISSUE THREAD

**Issue by cpage-pivotal** [ObsessiveCompulsiveAudiophile/AudysseyOne#1]
[Heights.ady.json](https://github.com/ObsessiveCompulsiveAudiophile/AudysseyOne/files/14617059/Heights.ady.json)

I've attached my .ady file. Running AudysseyOne with this file gives me some odd crossover settings:
```
6:34:51 PM [INFO] Pinpointing optimal crossover frequencies...
6:34:51 PM [INFO] Crossover frequency for C is set @ 40Hz
6:34:51 PM [INFO] Crossover frequency for FHL is set @ FULL RANGE <= 20Hz
6:34:51 PM [INFO] Crossover frequency for FHR is set @ 180Hz
6:34:52 PM [INFO] Crossover frequency for FL is set @ FULL RANGE <= 20Hz
6:34:52 PM [INFO] Crossover frequency for FR is set @ FULL RANGE <= 20Hz
6:34:52 PM [INFO] Crossover frequency for SLA is set @ FULL RANGE <= 20Hz
6:34:52 PM [INFO] Crossover frequency for SRA is set @ FULL RANGE <= 20Hz
```

For example, here's an ady measurement for my surround right speaker: 
<img width="1358" alt="image" src="https://github.com/ObsessiveCompulsiveAudiophile/AudysseyOne/assets/10927227/a8d7125c-9ad1-46a3-aa1c-2954bca42567">

It's obviously struggling below 60 hz, while my subwoofer is confident in that range:

<img width="1385" alt="image" src="https://github.com/ObsessiveCompulsiveAudiophile/AudysseyOne/assets/10927227/d1fb8862-a1bb-4bae-8066-c5d6252035b5">

It doesn't seem logical that AudysseyOne would recommend FULL RANGE crossover for these speakers. Is this a bug?

**Comment by ObsessiveCompulsiveAudiophile** [comment]
I am trying to understand how is your front height speakers, especially the right one is going flat all the way down to 9Hz!!!

Subwoofer also has a huge dip between 55Hz and 150Hz. You should consider moving it elsewhere in the room.

You might wanna re-run Audyssey calibration and use closer microphone positions. 10-15 cm around the MLP is enough. They seem to be more than 1.2m apart.

If you can get rid of the SW dip below 80Hz, you could crossover every speaker at 80Hz which is THX standard since your speakers have very different bass responses.

**Comment by cpage-pivotal** [comment]
I agree the front height speaker is surprising. I went back and took a fresh set of measurements through Audyssey to double-check, and got the same result. These are my height speakers: https://us.kef.com/products/ci200rr. They claim to extend to 33hz (+-6db), and the front right is in a corner, so it may be getting some room gain.

I re-ran AudysseyOne with the fresh measurements, and got nearly the same crossover recommendations. For the Front Height Right, which extends low, it recommended a 180hz crossover! And for the fronts and surrounds, it recommended full-range.

Again, looking at the curves for the surround speakers, I can't see why it would conclude they were full-range. I'm happy to set the crossovers myself to 80hz, but I'm wondering if the algorithm could be improved.

**Comment by ObsessiveCompulsiveAudiophile** [comment]
If both pairs show same very high XO frequencies then they might really be
crossed over that high as they are both rolling off by 24dB/octave at that
frequency and any lower XO level will cause a NULL response between these
frequencies but with larger speakers, if one speaker shows up oddly high XO
freq and the other doesn't, the app will xo them both at the lower freq
anyway. That's why I left it as is so that the user can see, one of the
speakers have a major dip and maybe take physical measures to fix it. But
it's not giving confidence to the user to see such differences between the
pairs and you have a point. I will look into what could be done about that.
